src/lsp/LSPClient.js

```diff
@@ -204,19 +204,28 @@ export default class LSPClient {
     }
 
     const str = decoder_utf8.decode(uint8);
-    return JSON.parse(str);
+    try {
+      return JSON.parse(str);
+    } catch (err) {
+      await this.stop();
```

Comment:
Stopping the LSP on an error is a good idea, but not re-throwing the error could potentially make debugging harder?